src/tracepoint/tracepoint.bpf.c

```diff
@@ -0,0 +1,7642 @@
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_endian.h>
+#include "event.h"
+#include "map.h"
+
+#define ALLOW 0
+#define BLOCK 1
+#define LOGGING 2
+
+char LICENSE[] SEC("license") = "GPL";
+
+static __always_inline struct current_task get_task_struct() {
+    struct current_task ct = {};
+
+    struct task_struct *task = (struct task_struct *)bpf_get_current_task();
+    if (task == NULL) {
+        bpf_printk("failed to get cur task\n");
+        return ct;
+    }
+
+    __u64 pid_tgid = bpf_get_current_pid_tgid();
+    ct.pid = pid_tgid >> 32;
+    ct.tid = (__u32)pid_tgid;
+    ct.ns_id = BPF_CORE_READ(task, nsproxy, pid_ns_for_children, ns.inum);
+
+    return ct;
+}
+
+// network events related tracepoints
+SEC("tracepoint/syscalls/sys_enter_socket")
+int trace_sys_enter_socket(struct trace_event_raw_sys_enter *ctx) {
+    __s32 event_id = SYS_ENTER_SOCKET;
+
+    __s32 domain = BPF_CORE_READ(ctx, args[0]);
+    __s32 type = BPF_CORE_READ(ctx, args[1]);
+    __s32 protocol = BPF_CORE_READ(ctx, args[2]);
+
+    struct current_task ct = get_task_struct();
+
+    struct event_key event_key = {
+        .ns_id = ct.ns_id,
+        .event_id = event_id,
+    };
+
+    __u32 *watched = bpf_map_lookup_elem(&event_policy_map, &event_key);
+    if (watched) {
+        if (*watched == LOGGING) {
```

Comment:
```suggestion
        if (*watched == LOGGING) {
```

애초에 watched라는 조건이 있다는게 위 한줄로 가능하지않을까요? 값을 비교할 수 있다는게 애초에 watched는 true니까요

Reply:
커널은 해당 검증이 안되는군요 한 수 배웠습니다!